_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -2302,6 +2302,30 @@ mfxStatus ImplementationAvc::AsyncRoutine(mfxBitstream * bs)
 
             if (newTask.m_picStruct[ENC] == MFX_PICSTRUCT_FIELD_BFF)
                 std::swap(newTask.m_type.top, newTask.m_type.bot);
+
+            mfxU64 timeStamp = newTask.m_timeStamp;
+            // sort the time stamp if the input frame is in encoded order.
+            if (m_timeStamps.size() && (timeStamp != static_cast<mfxU64>(MFX_TIMESTAMP_UNKNOWN)))
```

Comment:
There is not only to compare two value "mfxU64", but also to deal with the time stamp wrapping case. It is implemented through static_cast the delta from mfxU64 to mfxI64.  For example, the timestamp "1" should insert after the "0xfffffffffffffffff".    Your code doesn't work for time stamp wrapping case.

Reply:
find_if would simplify the code, whatever logic is applied.
Anyway the logic is still not clear when input contains both valid and invalid pts.
Invalid pts are put at the end, but new valid pts are inserted before bigger valid if found, or at the end, even after all invalid pts. Is it supposed?

Reply:
Agree. Let's use the find_if to simplify the codes.
The principle of the logic: 1) The valid pts is in the increasing order. 2) the invalid pts is put to end.
It can make sure that 1) the frame DTS is increasing 2) DTS <=PTS for the individual frame.